osu.Game/Online/Rooms/Room.cs

```diff
@@ -264,6 +265,18 @@ public RoomAvailability Availability
             set => SetField(ref availability, value);
         }
 
+        [OnDeserialized]
+        private void onDeserialised(StreamingContext context)
```

Comment:
Not 100% on fixing this using newtonsoft magic. For one thing, I was surprised that this did anything for unit testing, upon which I stumbled upon

https://github.com/ppy/osu/blob/2c0140f8651dd773d8acea7e8e07c102217e89ea/osu.Game/Tests/Visual/OnlinePlay/TestRoomRequestsHandler.cs#L288-L294

and then

https://github.com/ppy/osu/blob/2c0140f8651dd773d8acea7e8e07c102217e89ea/osu.Game/Online/Rooms/PlaylistItem.cs#L46-L68

upon reading which I concluded that maybe this is *comparably* not that bad anymore and at least it has test coverage so I'm prolly gonna look away...